plugins/PRAplugin/src/MCSSolver.py

```diff
@@ -0,0 +1,146 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on June 24, 2020
+
+@author: mandd
+"""
+
+from __future__ import division, print_function , unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default', DeprecationWarning)
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import itertools
+import math
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from PluginsBaseClasses.ExternalModelPluginBase import ExternalModelPluginBase
+from PostProcessors.MCSimporter import MCSreader
+#Internal Modules End-----------------------------------------------------------
+
+class MCSSolver(ExternalModelPluginBase):
+  """
+    This class is designed to create a MCS solver model
+  """
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    ExternalModelPluginBase.__init__(self)
+
+  def _readMoreXML(self, container, xmlNode):
+    """
+      Method to read the portion of the XML that belongs to the MCS solver model
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, xmlNode, xml.etree.ElementTree.Element, XML node that needs to be read
+      @ Out, None
+    """
+    container.filename = None
+    container.mapping    = {}
+    container.InvMapping = {}
+
+    for child in xmlNode:
+      if child.tag == 'topEventID':
+        container.topEventID = child.text.strip()
+      elif child.tag == 'solverOrder':
+        self.solverOrder = int(child.text.strip())
+      elif child.tag == 'map':
+        container.mapping[child.get('var')]      = child.text.strip()
+        container.InvMapping[child.text.strip()] = child.get('var')
+      elif child.tag == 'variables':
+        variables = [str(var.strip()) for var in child.text.split(",")]
+      else:
+        raise IOError("MCSSolver: xml node " + str(child.tag) + " is not allowed")
+
+  def initialize(self, container, runInfoDict, inputFiles):
+    """
+      Method to initialize this plugin
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, runInfoDict, dict, dictionary containing all the RunInfo parameters (XML node <RunInfo>)
+      @ In, inputFiles, list, list of input files (if any)
+      @ Out, None
+    """
+    pass
+
+  def createNewInput(self, container, inputs, samplerType, **Kwargs):
+    """
+      This function has been added for this model in order to be able to create a FTstructure from multiple files
+      @ In, myInput, list, the inputs (list) to start from to generate the new one
```

Comment:
update docstring, myInput is not listed as input param, in addition, container and inputs are missing for docstring.

Reply:
fixed